File: leetcode/easy/205.swift
class Solution {
    func isIsomorphic(_ s: String, _ t: String) -> Bool {
        let sstr = Array(s)
        let tstr = Array(t)
        var m = [Character: Character]()
        
        for i in 0..<sstr.count {
            if !m.keys.contains(sstr[i]) {
                if !m.values.contains(tstr[i]) {
                    m[sstr[i]] = tstr[i]
                } else { return false }
            } else if m[sstr[i]] != tstr[i] {
                return false
            }
        }
        return true
    }
}

File: leetcode/easy/205.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        int fromTo[200] = { 0, };
        int used[200] = { 0, };
        for (int i = 0; i < s.size(); i++) {
            int sc = s[i];
            int tc = t[i];
            if (fromTo[sc] == 0) {
                if (used[tc] == 1) { return false; }
                fromTo[sc] = tc;
                used[tc] = 1;
            } else {
                if (fromTo[sc] != tc) { return false; }
            }
        }
        return true;
    }
};
